src/main/java/org/apache/sysml/runtime/matrix/data/SparseBlockCSR.java

```diff
@@ -848,7 +848,57 @@ public String toString() {
 		
 		return sb.toString();
 	}
-	
+
+//	@Override
+	public boolean checkValidity(int rlen, int clen, boolean strict) {
+		//1. correct meta data
+		if( rlen < 0 || clen < 0 ) {
+			throw new RuntimeException("Invalid block dimensions: "+rlen+" "+clen);
+		}
+
+		//2. correct array lengths
+		int nnz = _size;
+		long esize = estimateMemory(rlen, clen, MatrixBlock.SPARSITY_TURN_POINT);
+		if(_size < esize && _ptr.length > rlen+1 && _values.length  >= nnz && _indexes.length >= nnz ) {
```

Comment:
Please don't use the memory estimate which is in number of bytes not number of elements. The other checks are good (see above).